backend/api/src/cast-poll-vote.ts

```diff
@@ -0,0 +1,59 @@
+import { createSupabaseDirectClient } from 'shared/supabase/init'
+import { log } from 'shared/utils'
+import { z } from 'zod'
+import { APIError, authEndpoint, validate } from './helpers'
+import * as admin from 'firebase-admin'
+import { PollOption } from 'common/poll-option'
+import { Contract, PollContract } from 'common/contract'
+
+const schema = z.object({
+  contractId: z.string(),
+  voteId: z.string(),
+})
+
+export const castpollvote = authEndpoint(async (req, auth) => {
+  const { contractId, voteId } = validate(schema, req.body)
+  const pg = createSupabaseDirectClient()
+
+  const voters = await pg.manyOrNone(
```

Comment:
We could use a pg promise transaction:

(example)
```
db.tx(async t => {
    await t.none('UPDATE users SET active = $1 WHERE id = $2', [true, 123]);
    await t.one('INSERT INTO audit(entity, id) VALUES($1, $2) RETURNING id', ['users', 123]);
})
``` 

If we both 1. read the voters and 2. insert a voter, within a transaction, that could solve some double-vote race conditions.

https://github.com/vitaly-t/pg-promise#transactions